app/models/bulkrax/oai_adventist_qdc_entry.rb

```diff
@@ -7,5 +7,23 @@ class OaiAdventistQdcEntry < OaiQualifiedDcEntry
     def add_thumbnail_url
       true
     end
+
+    def collections_created?
+      true
+    end
+
+    def find_collection_ids
+      return self.collection_ids if collections_created?
```

Comment:
The guard condition `return self.collection_ids if collections_created?` is serving as memoization (e.g. caching the results of an expensive operation).  However, it's not a self-contained memoization (e.g. the method is fully in charge of managing the memoization).  It instead relies on the previous implementation of `#collections_created?`.

With the change to `#collections_created?` we end up always returning `[]` for `#find_collection_ids`; because the memoization mechanism is dependent on external methods.

The impact of the above code as written is that we'll never create collections based on the record's `setSpec` XML elements.

What we need to do is modify the method so that we internalize the memoization:

```ruby
def find_collection_ids
  # Using memoization to cache what could be expensive queries.
  return self.collection_ids if defined?(@found_collection_ids)

  if sets.blank? || parser.collection_name != 'all'
    collection = find_collection(importerexporter.unique_collection_identifier(parser.collection_name))
    self.collection_ids << collection.id if collection.present? && !self.collection_ids.include?(collection.id)
  else # All - collections should exist for all sets
    sets.each do |set|
      c = find_collection(importerexporter.unique_collection_identifier(set.content))
      self.collection_ids << c.id if c.present? && !self.collection_ids.include?(c.id)
    end
  end
  
  # We've completed our expensive queries, now let's make sure we don't need to do this again.
  @found_collection_ids = true

  return self.collection_ids
end
```

Reply:
ah, we misunderstood the instructions of the issue. If that's what case, why are we adding the collections_created method? And why should it always return true? Perhaps bulkrax is calling that method somewhere then? I'll go look.